### PR TITLE
logging(gitstart): migrate internal/database to lib/log

### PR DESCRIPTION
**Pull request by gitstart-sourcegraph**
> ## Problem statement
> `internal/database` is a shared library that still uses log15. We should migrate this package to use lib/log instead.
> 
> ## Success criteria
> All 30 usages of log15 in `internal/database` are replaced with calls to an appropriately scoped `lib/log` logger.
> 
> https://sourcegraph.com/search?q=context:global+repo:%5Egithub%5C.com/sourcegraph/sourcegraph%24+file:%5Einternal/database+log15&patternType=literal
> 
> ## Implementation details
> Callsites provide loggers to the package (no logger should be instantiated within the package)
> 
> * In struct functions, all loggers should be set to the struct and provided.
> * In helper functions, all loggers should be provided as a parameter.
> 
> ## Time estimate
> * Pull requests with ~450 lines changed should take **48 hours** at maximum. Ping the reviewer in the spec pull request if time-consuming changes are required.
> * Split the work into multiple pull requests if the total diff is bigger than 450 lines of code.

### Refs
[Sourcegraph Issue](https://github.com/sourcegraph/sourcegraph/issues/36387)
[GitStart Issue](https://app.gitstart.com/clients/sourcegraph/tickets/SG-36387)

## Test plan
- A good way to test this PR is just to run the branch locally, and open [the issue link](https://github.com/sourcegraph/sourcegraph/issues/36387), compare the changes on the PR and the listed issues. All of them must be fixed

### CODE REVIEW

File: go.sum [hunk 1]
```diff
@@ -92,6 +92,8 @@ contrib.go.opencensus.io/resource v0.1.1/go.mod h1:F361eGI91LCmW1I/Saf+rX0+OFcig
 dmitri.shuralyov.com/gpu/mtl v0.0.0-20190408044501-666a987793e9/go.mod h1:H6x//7gZCb22OMCxBHrMx7a5I7Hp++hsVxbQ4BYO7hU=
 git.apache.org/thrift.git v0.0.0-20180902110319-2566ecd5d999/go.mod h1:fPE2ZNJGynbRyZ4dJvy6G277gSllfV2HJqblrnkyeyg=
 git.apache.org/thrift.git v0.12.0/go.mod h1:fPE2ZNJGynbRyZ4dJvy6G277gSllfV2HJqblrnkyeyg=
+github.com/766b/go-outliner v0.0.0-20180511142203-fc6edecdadd7 h1:cJXisB2yAM61AzMutv7X+KM8F3xVLxGH99S8VmaSlps=
+github.com/766b/go-outliner v0.0.0-20180511142203-fc6edecdadd7/go.mod h1:1SzhThoS5lcKfE4IFOLQJ04WCmFpaAiPe8H9yqXyYSU=
```
Comment (gitstart-sourcegraph) [inline go.sum:96]: Why this?

File: internal/database/migration/runner/logger.go [hunk 2]
```diff
@@ -3,22 +3,11 @@ package runner
...
-func EnableLogging() {
-	logger = log15.Root()
-}
+func EnableLogging() {}
 
-func DisableLogging() {
-	logger = log15.New()
-	logger.SetHandler(log15.DiscardHandler())
-}
+func DisableLogging() {}
```
Comment (gitstart-sourcegraph) [inline internal/database/migration/runner/logger.go:10]: Can we remove these functions and instances they are called?

File: internal/database/migration/runner/runner.go [hunk 3]
```diff
@@ -465,12 +464,12 @@ func renderIndexStatus(progress storetypes.IndexStatus) (logPairs []any) {
...
-		"lockers", fmt.Sprintf("%d of %d", progress.LockersDone, progress.LockersTotal),
-		"blocks", fmt.Sprintf("%d of %d", progress.BlocksDone, progress.BlocksTotal),
-		"tuples", fmt.Sprintf("%d of %d", progress.TuplesDone, progress.TuplesTotal),
+	return []log.Field{
+		log.Bool("in-progress", false),
+		log.String("phase", *progress.Phase),
+		log.String("phases", fmt.Sprintf("%d of %d", index, len(storetypes.CreateIndexConcurrentlyPhases))),
+		log.String("lockers", fmt.Sprintf("%d of %d", progress.LockersDone, progress.LockersTotal)),
+		log.String("blocks", fmt.Sprintf("%d of %d", progress.BlocksDone, progress.BlocksTotal)),
+		log.String("tuples", fmt.Sprintf("%d of %d", progress.TuplesDone, progress.TuplesTotal)),
```
Comment (gitstart-sourcegraph) [inline internal/database/migration/runner/runner.go:473]: Use structural logging

File: internal/database/migration/runner/run.go [hunk 4]
```diff
@@ -84,50 +85,37 @@ func (r *Runner) runSchema(
...
-		"pendingVersions", extractIDs(byState.pending),
-		"failedVersions", extractIDs(byState.failed),
-		"targetDefinitions", extractIDs(definitions),
-	)
-	logger.Info(
-		"Checking for active migrations",
-		"schema", schemaContext.schema.Name,
-	)
+	logger.Warn("Schema not in expected state", log.Ints("targetDefinitions", extractIDs(definitions)))
+	slogger.Info("Checking for active migrations")
```
Comment (burmudar) [inline internal/database/migration/runner/run.go:118]: `targetDefinitions` is missing here

File: internal/database/migration/runner/run.go [hunk 5]
```diff
@@ -84,50 +85,37 @@ func (r *Runner) runSchema(
...
-		"pendingVersions", extractIDs(byState.pending),
-		"failedVersions", extractIDs(byState.failed),
+	logger := r.logger.With(
+		log.String("schema", schemaContext.schema.Name),
+		log.Ints("appliedVersions", extractIDs(byState.applied)),
+		log.Ints("pendingVersions", extractIDs(byState.pending)),
+		log.Ints("failedVersions", extractIDs(byState.failed)),
 	)
 
+	slogger := r.logger.With(log.String("schema", schemaContext.schema.Name))
```
Comment (burmudar) [inline internal/database/migration/runner/run.go:95]: it is a bit confusing to have two different loggers here

File: internal/database/migration/runner/runner.go [hunk 6]
```diff
@@ -27,12 +29,16 @@ func NewRunner(storeFactories map[string]StoreFactory) *Runner {
 
 func NewRunnerWithSchemas(storeFactories map[string]StoreFactory, schemas []*schemas.Schema) *Runner {
 	return &Runner{
+		logger:         log.Scoped("NewRunnerWithSchemas", "create new runner with schema"),
```
Comment (burmudar) [inline internal/database/migration/runner/runner.go:32]: Think this should just be `log.Scoped("Runner", "")`

File: internal/database/migration/runner/runner.go [hunk 7]
```diff
@@ -431,15 +430,15 @@ func getAndLogIndexStatus(ctx context.Context, schemaContext schemaContext, tabl
...
+	schemaContext.logger.Info(
 		"Checked progress of index creation",
 		append(
-			[]any{
-				"schema", schemaContext.schema.Name,
-				"tableName", tableName,
-				"indexName", indexName,
-				"exists", exists,
-				"isValid", indexStatus.IsValid,
+			[]log.Field{
```
Comment (burmudar) [inline internal/database/migration/runner/runner.go:436]: might be better to use `log.Object` here instead

File: internal/database/migration/runner/runner.go [hunk 8]
```diff
@@ -465,12 +464,12 @@ func renderIndexStatus(progress storetypes.IndexStatus) (logPairs []any) {
...
 	}
 
-	return []any{
-		"in-progress", true,
-		"phase", *progress.Phase,
-		"phases", fmt.Sprintf("%d of %d", index, len(storetypes.CreateIndexConcurrentlyPhases)),
-		"lockers", fmt.Sprintf("%d of %d", progress.LockersDone, progress.LockersTotal),
-		"blocks", fmt.Sprintf("%d of %d", progress.BlocksDone, progress.BlocksTotal),
-		"tuples", fmt.Sprintf("%d of %d", progress.TuplesDone, progress.TuplesTotal),
+	return []log.Field{
```
Comment (burmudar) [inline internal/database/migration/runner/runner.go:467]: might be better to use `log.Object` here instead

File: internal/database/dbcache/cached_indexable_repos.go [hunk 9]
```diff
@@ -38,6 +37,7 @@ var globalReposCache = reposCache{}
 
 func NewIndexableReposLister(store database.RepoStore) *IndexableReposLister {
 	return &IndexableReposLister{
+		logger:     log.Scoped("NewIndexableReposLister", "List indexable repos"),
```
Comment (burmudar) [inline internal/database/dbcache/cached_indexable_repos.go:40]: Scoped name should be according to the struct/component name ie. `IndexableReposLister`

File: internal/database/dbconn/config.go [hunk 10]
```diff
@@ -21,6 +21,7 @@ var (
 // buildConfig takes either a Postgres connection string or connection URI,
 // parses it, and returns a config with additional parameters.
 func buildConfig(dataSource, app string) (*pgx.ConnConfig, error) {
```
Comment (burmudar) [inline internal/database/dbconn/config.go:23]: I think  a logger should be passed in as a param here instead

File: internal/database/external_services.go [hunk 11]
```diff
@@ -152,7 +153,10 @@ func (e *externalServiceStore) copy() *externalServiceStore {
 
 // ExternalServicesWith instantiates and returns a new ExternalServicesStore with prepared statements.
 func ExternalServicesWith(other basestore.ShareableStore) ExternalServiceStore {
-	return &externalServiceStore{Store: basestore.NewWithHandle(other.Handle())}
+	return &externalServiceStore{
+		logger: log.Scoped("External service store", "ExternalServicesWith instantiates and returns a new ExternalServicesStore with prepared statements"),
```
Comment (burmudar) [inline internal/database/external_services.go:157]: use the same case as the struct for the scope name `externalServiceStore`. Also the description should be empty in this case and not the comment to the method.

File: internal/database/migration/runner/run.go (hunk 5 above)
Comment (gitstart-sourcegraph) [reply internal/database/migration/runner/run.go:95]: The structural logging values for `logger` and `slogger` are different but are reused in multiple places. So they were scopped to their appropriate `logger` and used accordingly

File: internal/database/repos.go [hunk 12]
```diff
@@ -413,6 +413,8 @@ func scanRepo(rows *sql.Rows, r *types.Repo) (err error) {
 	var metadata json.RawMessage
 	var blocked dbutil.NullJSONRawMessage
 
+	logger := log.Scoped("scanRepo", "scan the repo")
```
Comment (burmudar) [inline internal/database/repos.go:416]: If we look where `scanRepo` [is being used](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@36c320a88aa7594663ede8ce0cf3b5e97f0d8f69/-/blob/internal/database/repos.go?L784:13)
I think we should rather define the logger on the original `database/repos.go` struct and pass it in here as a parameter.

File: internal/database/dbconn/config.go (hunk 10 above)
Comment (gitstart-sourcegraph) [reply internal/database/dbconn/config.go:23]: At what point do we pass logger as a parameter or create a new scope? Is there a rule to follow?

File: internal/database/migration/runner/run.go [hunk 13]
```diff
@@ -84,50 +85,37 @@ func (r *Runner) runSchema(
...
-		"schema", schemaContext.schema.Name,
-		"appliedVersions", extractIDs(byState.applied),
-		"pendingVersions", extractIDs(byState.pending),
-		"failedVersions", extractIDs(byState.failed),
+	logger := r.logger.With(
+		log.String("schema", schemaContext.schema.Name),
+		log.Ints("appliedVersions", extractIDs(byState.applied)),
+		log.Ints("pendingVersions", extractIDs(byState.pending)),
+		log.Ints("failedVersions", extractIDs(byState.failed)),
 	)
```
Comment (bobheadxi) [inline internal/database/migration/runner/run.go:93]: This migration is not quite correct - we are creating an _info_ log entry with the fields, so you want:

```
logger := r.logger.With(
		log.String("schema", schemaContext.schema.Name),
	)

logger.Info("Checked current schema state",
		log.Ints("appliedVersions", extractIDs(byState.applied)),
		log.Ints("pendingVersions", extractIDs(byState.pending)),
		log.Ints("failedVersions", extractIDs(byState.failed)))
```

File: internal/database/repos.go [hunk 14]
```diff
@@ -408,13 +414,11 @@ var repoColumns = []string{
 	"repo.blocked",
 }
 
-func scanRepo(rows *sql.Rows, r *types.Repo) (err error) {
+func scanRepo(rows *sql.Rows, r *types.Repo, logger log.Logger) (err error) {
```
Comment (burmudar) [inline internal/database/repos.go:417]: `logger` should be the first param here. If `context` is first, then logger should be second

File: internal/database/repos.go [hunk 15]
```diff
@@ -88,22 +88,28 @@ var _ RepoStore = (*repoStore)(nil)
...
+	logger log.Logger
 	*basestore.Store
 }
 
 // ReposWith instantiates and returns a new RepoStore using the other
 // store handle.
 func ReposWith(other basestore.ShareableStore) RepoStore {
-	return &repoStore{Store: basestore.NewWithHandle(other.Handle())}
+	return &repoStore{
+		logger: log.Scoped("ReposWith", ""),
```
Comment (burmudar) [inline internal/database/repos.go:101]: the scoped name should be the struct name `repoStore`

File: internal/database/migration/runner/runner.go [hunk 16]
```diff
@@ -432,28 +432,24 @@ func getAndLogIndexStatus(ctx context.Context, schemaContext schemaContext, tabl
...
 // renderIndexStatus returns a slice of interface pairs describing the given index status for use in a
 // call to logger. If the index is currently being created, the progress of the create operation will be
 // summarized.
-func renderIndexStatus(progress storetypes.IndexStatus) []log.Field {
+func renderIndexStatus(progress storetypes.IndexStatus) log.Field {
 	if progress.Phase == nil {
-		return []log.Field{
-			log.Bool("in-progress", false),
-		}
+		return log.Object("progress-hase", log.Bool("in-progress", false))
```
Comment (burmudar) [inline internal/database/migration/runner/runner.go:452]: I think the object name should have the same object name as on line 463

File: internal/database/dbconn/config_test.go [hunk 17]
```diff
@@ -2,9 +2,12 @@ package dbconn
 
 import (
 	"testing"
+
+	"github.com/sourcegraph/sourcegraph/lib/log/logtest"
 )
 
 func TestBuildConfig(t *testing.T) {
+	logger := logtest.Scoped(t)
```
Comment (bobheadxi) [inline internal/database/dbconn/config_test.go:10]: This should be the tightest scope possible, i.e. on line 51

File: internal/database/external_services.go [hunk 18]
```diff
@@ -152,7 +153,10 @@ func (e *externalServiceStore) copy() *externalServiceStore {
 
 // ExternalServicesWith instantiates and returns a new ExternalServicesStore with prepared statements.
 func ExternalServicesWith(other basestore.ShareableStore) ExternalServiceStore {
-	return &externalServiceStore{Store: basestore.NewWithHandle(other.Handle())}
+	return &externalServiceStore{
+		logger: log.Scoped("externalServiceStore", ""),
```
Comment (bobheadxi) [inline internal/database/external_services.go:157]: Same here, this should accept a logger - as outlined in the issue requirements, there should be no instantiation of loggers in this package. Please double check all instances of `log.Scoped()` in this PR to make sure that all usages are only `logger.Scoped()`, _not_ logger instantiations

**Review by bobheadxi — CHANGES_REQUESTED** [review]
Reminder of the requirements outlined:

> Callsites provide loggers to the package (**no logger should be instantiated within the package**)
>
> - In struct functions, all loggers should be set to the struct and provided.
> - In helper functions, all loggers should be provided as a parameter.

File: internal/database/dbconn/connect.go [hunk 19]
```diff
@@ -20,8 +20,8 @@ import (
 //
 // Note: github.com/jackc/pgx parses the environment as well. This function will also use the value
 // of PGDATASOURCE if supplied and dataSource is the empty string.
-func ConnectInternal(dsn, appName, dbName string) (_ *sql.DB, err error) {
-	logger := log.Scoped("ConnectInternal", "connects to the given data source and return the handle")
+func ConnectInternal(logger log.Logger, dsn, appName, dbName string) (_ *sql.DB, err error) {
+	// logger := log.Scoped("ConnectInternal", "connects to the given data source and return the handle")
```
Comment (gitstart-sourcegraph) [inline internal/database/dbconn/connect.go:24]: Please remove commented code

File: internal/database/dbtest/dbtest.go [hunk 20]
```diff
@@ -185,8 +187,9 @@ func wdHash() string {
 }
 
 func dbConn(t testing.TB, cfg *url.URL, schemas ...*schemas.Schema) *sql.DB {
+	logger := log.Scoped("dbConn", "connects a new test db")
```
Comment (gitstart-sourcegraph) [inline internal/database/dbtest/dbtest.go:190]: This is a test file. Shouldn't we be using the `logtest`

File: internal/database/migration/cliutil/addlog.go [hunk 21]
```diff
@@ -4,14 +4,14 @@ import (
...
-	"github.com/inconshreveable/log15"
 	"github.com/urfave/cli/v2"
 
 	"github.com/sourcegraph/sourcegraph/internal/database/migration/definition"
+	"github.com/sourcegraph/sourcegraph/lib/log"
 	"github.com/sourcegraph/sourcegraph/lib/output"
 )
 
-func AddLog(commandName string, factory RunnerFactory, outFactory OutputFactory) *cli.Command {
+func AddLog(logger log.Logger, commandName string, factory RunnerFactory, outFactory OutputFactory) *cli.Command {
```
Comment (bobheadxi) [inline internal/database/migration/cliutil/addlog.go:14]: Don't pass this logger in here, just create a logger in the action below

File: internal/database/migration/runner/validate.go [hunk 22]
```diff
@@ -28,37 +29,28 @@ func (r *Runner) validateSchema(ctx context.Context, schemaContext schemaContext
...
-		"appliedVersions", extractIDs(byState.applied),
-		"pendingVersions", extractIDs(byState.pending),
-		"failedVersions", extractIDs(byState.failed),
-		"targetDefinitions", extractIDs(definitions),
-	)
-	logger.Info(
-		"Checking for active migrations",
-		"schema", schemaContext.schema.Name,
-	)
+	logger.Warn("Schema not in expected state")
```
Comment (bobheadxi) [inline internal/database/migration/runner/validate.go:51]:
```suggestion
	slogger.Warn("Schema not in expected state")
```

File: internal/database/security_event_logs.go [hunk 23]
```diff
@@ -66,6 +65,9 @@ type SecurityEventLogsStore interface {
 }
 
 type securityEventLogsStore struct {
+	// logger is a standardized, strongly-typed, and structured logging interface
+	// Production output from this logger (SRC_LOG_FORMAT=json) complies with the OpenTelemetry log data model
```
Comment (bobheadxi) [inline internal/database/security_event_logs.go:69]: I think we can remove all of these comments (in the other fields as well)

**Review by bobheadxi — CHANGES_REQUESTED** [review]

File: internal/database/users.go [hunk 24]
```diff
@@ -87,23 +87,37 @@ type UserStore interface {
...
+	logger log.Logger
 	*basestore.Store
 }
 
 var _ UserStore = (*userStore)(nil)
 
+// Users instantiates and returns a new RepoStore with prepared statements.
+func Users(db dbutil.DB) UserStore {
+	return &userStore{
+		logger: log.Scoped("userStore", ""),
```
Comment (bobheadxi) [inline internal/database/users.go:101]: initialization should not happen here

Comment (gitstart-sourcegraph) [reply internal/database/users.go:101]: Do we also need to pass in the logger into the `Users` func?

Comment (bobheadxi) [reply internal/database/users.go:101]: Yes, quoting from the issue description:

> no logger should be instantiated within the package

The package being `internal/database`, which this code is within

File: cmd/frontend/internal/cli/serve_cmd.go [hunk 25]
```diff
@@ -107,13 +108,13 @@ func defaultExternalURL(nginxAddr, httpAddr string) *url.URL {
 
 // InitDB initializes and returns the global database connection and sets the
 // version of the frontend in our versions table.
-func InitDB() (*sql.DB, error) {
+func InitDB(logger sglog.Logger) (*sql.DB, error) {
 	sqlDB, err := connections.EnsureNewFrontendDB("", "frontend", &observation.TestContext)
```
Comment (jhchabran) [inline cmd/frontend/internal/cli/serve_cmd.go:112]: Following up on https://sourcegraph.slack.com/archives/C03FWK7UZMK/p1655460973587039 

The `EnsureNewFrontendDB` function will pull its logger from the `observation.Context` here, whose logger is `nil` at this stage.

Comment (jhchabran) [reply cmd/frontend/internal/cli/serve_cmd.go:112]: Quick fix: 

```go
func InitDB(logger sglog.Logger) (*sql.DB, error) {
	obsCtx := observation.TestContext
	obsCtx.Logger = logger
	sqlDB, err := connections.EnsureNewFrontendDB("", "frontend", &obsCtx)
```

File: cmd/frontend/graphqlbackend/webhook_logs.go [hunk 26]
```diff
@@ -98,6 +100,9 @@ func (r *schemaResolver) WebhookLogs(ctx context.Context, args *globalWebhookLog
 }
 
 type webhookLogConnectionResolver struct {
+	// logger is a standardized, strongly-typed, and structured logging interface
+	// Production output from this logger (SRC_LOG_FORMAT=json) complies with the OpenTelemetry log data model
```
Comment (bobheadxi) [inline cmd/frontend/graphqlbackend/webhook_logs.go:104]: Minor note - these docstrings are never required, don't worry about adding them :)

File: cmd/frontend/internal/app/ui/handlers.go [hunk 27]
```diff
@@ -127,6 +129,7 @@ var mockNewCommon func(w http.ResponseWriter, r *http.Request, title string, ser
 // In the case of a repository that is cloning, a Common data structure is
 // returned but it has an incomplete RevSpec.
 func newCommon(w http.ResponseWriter, r *http.Request, db database.DB, title string, indexed bool, serveError serveErrorHandler) (*Common, error) {
+	logger := sglog.Scoped("server error handler", "")
```
Comment (bobheadxi) [inline cmd/frontend/internal/app/ui/handlers.go:131]:
```suggestion
	logger := sglog.Scoped("commonHandler", "")
```

File: cmd/frontend/internal/httpapi/httpapi.go [hunk 28]
```diff
@@ -123,6 +125,7 @@ func NewHandler(
 // This handler is not guaranteed to provide the same authorization checks as
 // public API handlers.
 func NewInternalHandler(m *mux.Router, db database.DB, schema *graphql.Schema, newCodeIntelUploadHandler enterprise.NewCodeIntelUploadHandler, newComputeStreamHandler enterprise.NewComputeStreamHandler, rateLimitWatcher graphqlbackend.LimitWatcher) http.Handler {
+	logger := sglog.Scoped("NewInternalHandler", "")
```
Comment (bobheadxi) [inline cmd/frontend/internal/httpapi/httpapi.go:128]:
```suggestion
	logger := sglog.Scoped("InternalHandler", "")
```

File: cmd/github-proxy/github-proxy.go [hunk 29]
```diff
@@ -31,8 +34,7 @@ import (
 	"github.com/sourcegraph/sourcegraph/internal/trace/ot"
 	"github.com/sourcegraph/sourcegraph/internal/tracer"
 	"github.com/sourcegraph/sourcegraph/internal/version"
-	"github.com/sourcegraph/sourcegraph/lib/log"
-	"github.com/sourcegraph/sourcegraph/lib/log/otfields"
+	// "github.com/sourcegraph/sourcegraph/lib/log/otfields"
```
Comment (bobheadxi) [inline cmd/github-proxy/github-proxy.go]:
```suggestion
```

File: doc/dev/how-to/add_logging.md [hunk 30]
```diff
@@ -33,7 +33,7 @@ Initialize `lib/log` package within your program's `main()` function, for exampl
 
 ```go
 import (
-  "github.com/sourcegraph/sourcegraph/lib/log"
+  "github.com/sourcegraph/log"
```
Comment (bobheadxi) [inline doc/dev/how-to/add_logging.md:39]: This diff should not be in your branch - maybe a botched merge?

File: doc/dev/how-to/add_logging.md [hunk 31]
```diff
@@ -287,4 +298,12 @@ func TestFooBar(t *testing.T) {
...
 
+<<<<<<< HEAD
+## Conventions
+
+* The first scope of the logger should be the name of the service and following the same naming of the service. In general, if the logger is initialized as described in [handling logging](#handling-logging) the name should be correct.
+* The logger parameter should either be immediately after `ctx context.Context`, or be the first parameter.
+* In some cases there might already be a `log` module imported. Use the alias `sglog` to refer to `lib/log`, for example `import sglog "github.com/sourcegraph/log"`.
+=======
 If you can provide a `Logger` instance, `logtest.NoOp()` can be used to silence output. Levels can also be adjusted using `(Logger).IncreaseLevel`.
+>>>>>>> 7f1aa182f1e3a0491f5a3b81f34c099db38cdd08
```
Comment (bobheadxi) [inline doc/dev/how-to/add_logging.md:309]: This file should have no diffs :)

File: internal/database/batch/batch.go [hunk 32]
```diff
@@ -147,7 +149,7 @@ func NewInserterWithReturn(
 		onConflictSuffix:     onConflictSuffix,
 		returningSuffix:      returningSuffix,
 		returningScanner:     returningScanner,
-		operations:           getOperations(),
+		operations:           getOperations(sglog.Scoped("Inserter", "")),
```
Comment (bobheadxi) [inline internal/database/batch/batch.go:152]: I think we should have this provided

**Comment by bobheadxi** [comment]
Overall LGTM, left some comments!

File: doc/dev/how-to/add_logging.md [hunk 33]
```diff
@@ -114,6 +113,18 @@ In general:
 Example:
 
 ```go
```
Comment (bobheadxi) [inline doc/dev/how-to/add_logging.md:115]: @gitstart-sourcegraph can we revert all changes to this file?

**Comment by bobheadxi** [comment]
Kicking off a `main-dry-run`: https://buildkite.com/sourcegraph/sourcegraph/builds?branch=main-dry-run%2Fcontractors%2FSG-36387

**Review by bobheadxi — APPROVED** [review]
CI looks green, and `sg start` seems healthy :) Thank you!

File: cmd/worker/internal/migrations/migrators/extsvc_webhook_migrator.go [hunk 34]
```diff
@@ -14,6 +16,9 @@ import (
 // has_webhooks field on external services based on the external service
 // configuration.
 type ExternalServiceWebhookMigrator struct {
+	// logger is a standardized, strongly-typed, and structured logging interface
+	// Production output from this logger (SRC_LOG_FORMAT=json) complies with the OpenTelemetry log data model
```
Comment (bobheadxi) [inline cmd/worker/internal/migrations/migrators/extsvc_webhook_migrator.go:20]: @gitstart-sourcegraph mentioned in another review comment that these comments should be removed, can you please remove them?

Comment (bobheadxi) [reply cmd/worker/internal/migrations/migrators/extsvc_webhook_migrator.go:20]: Here's all the places it should be removed from: https://sourcegraph.com/search?q=context:%40sourcegraph/all+%22logger+is+a+standardized%2C+strongly-typed%2C+and+structured+logging+interface%22+repo:github.com/sourcegraph/sourcegraph%24%40contractors/SG-36387&patternType=regexp

**Review by bobheadxi — CHANGES_REQUESTED** [review]
see review comment: https://github.com/sourcegraph/sourcegraph/pull/36466#discussion_r901981547

**Review by bobheadxi — APPROVED** [review]
Ah well, wasn't that big of a deal so I removed the comments myself in https://github.com/sourcegraph/sourcegraph/pull/36466/commits/8c532e9a2c7c607387ded61ff5b2f7a2a50ca0c2 :)

**Comment by bobheadxi** [comment]
@cla-bot check

**Pull request merged**